quint/src/quintParserFrontend.ts

```diff
@@ -97,7 +101,7 @@ export function parsePhase1(text: string, sourceLocation: string): ParseResult<P
     return left(errorMessages)
   } else {
     // walk through the AST and construct the IR
-    const listener = new ToIrListener(sourceLocation)
+    const listener = new ToIrListener(sourceLocation, idGen)
     ParseTreeWalker.DEFAULT.walk(listener as QuintListener, tree)
```

Comment:
Why make the id generator global but still have to thread it thru all the classes as a parameter? Seems a bit like the worst of both worlds :smile:

Reply:
I would prefer it to be a parameter in the most cases. But there a few places, where it is hard to carry the id generator around (e.g., runtime values). That's why the global variable is there.

Reply:
Also, we have to override the generator when doing tests. So we have the global generator to server as a common source of identifiers for various components. But the components themselves should be able to consume various generators.